codegen/builtin_fs.go

```diff
@@ -708,6 +707,34 @@ func (dp DockerPush) Call(ctx context.Context, cln *client.Client, ret Register,
 		}
 	}
 
+	dockerAPI, err := DockerAPI(ctx)
+	if dockerAPI.Moby {
+		// Return error only if dockerPush is using docker engine instead of buildkit.
+		if err != nil {
+			return Arg(ctx, 0).WithError(err)
+		}
+
+		exportFS.SolveOpts = append(exportFS.SolveOpts,
+			solver.WithPushMoby(ref),
+			solver.WithCallback(func(_ context.Context, resp *client.SolveResponse) error {
+				mw := MultiWriter(ctx)
+				if mw == nil {
+					return nil
+				}
+
+				pw := mw.WithPrefix("", false)
+				return progress.Wrap(fmt.Sprintf("pushing %s", ref), pw.Write, func(l progress.SubLogger) error {
```

Comment:
nit: could be simplified:

```suggestion
				return progress.Wrap("pushing "+ref, pw.Write, func(l progress.SubLogger) error {
```